src/processor/marketplace-events/agreement.ts

```diff
@@ -15,11 +15,11 @@ const handlers: HandlersObject<MarketplaceEvent, BaseEventProcessorOptions> = {
   async NewAgreement (event: MarketplaceEvent, options: BaseEventProcessorOptions): Promise<void> {
     const newAgreement = event.payload
 
-    await Agreement.upsert(newAgreement) // Agreement might already exist
+    const [agreement] = await Agreement.upsert(newAgreement) // Agreement might already exist
     logger.info(`Created new Agreement with ID ${newAgreement.agreementReference} for offer ${newAgreement.offerId}`)
 
     if (options.manager) {
-      await options.manager.pin(newAgreement.dataReference, parseInt(newAgreement.size))
+      await options.manager.pin(newAgreement.dataReference, agreement.size)
```

Comment:
This has been changed from `newAgreement` to `agreement` . Is that correct?

Reply:
Nope. It's just remove `parseInt`